src/management-system-v2/app/(dashboard)/[environmentId]/tasklist/tasklist.tsx

```diff
@@ -0,0 +1,448 @@
+              style={{ width: '100%' }}
+              onChange={(_, selectedUsers) => {
+                setUsersFilter(
+                  (selectedUsers as { label: string; value: string }[]).map((user) => user.value),
+                );
+              }}
+            ></Select>
+          ),
+        },
+      ],
+    },
+    {
+      type: 'divider',
+    },
+    {
+      key: '5',
+      type: 'group',
+      label: 'Groups',
+      children: [
+        {
+          key: '5-1',
+          label: (
+            <Select
+              mode="multiple"
+              allowClear
+              placeholder="Select Group(s)"
+              maxTagCount={5}
+              options={groups.map((group) => ({ label: group, value: group }))}
+              value={groupsFilter}
+              style={{ width: '100%' }}
+              onChange={(_, selectedGroups) => {
+                setGroupsFilter(
+                  (selectedGroups as { label: string; value: string }[]).map(
+                    (group) => group.value,
+                  ),
+                );
+              }}
+            ></Select>
+          ),
+        },
+      ],
+    },
+  ];
+
+  const sortValues = ['startTime', 'deadline', 'progress', 'priority', 'state'];
+
+  const sortDropdownItems: MenuProps['items'] = sortValues.map((sortValue, index) => {
+    return {
+      key: index,
+      label: (
+        <div
+          style={{ display: 'flex', alignItems: 'center' }}
+          onClick={() => {
+            if (selectedSortItem.value === sortValue) {
+              setSelectedSortItem({ ascending: !selectedSortItem.ascending, value: sortValue });
+            } else {
+              setSelectedSortItem({ ascending: true, value: sortValue });
+            }
+          }}
+        >
+          <span style={{ marginRight: '0.25rem' }}>{sortValue}</span>
+          {selectedSortItem.value === sortValue && selectedSortItem.ascending && (
+            <FaLongArrowAltUp></FaLongArrowAltUp>
+          )}
+          {selectedSortItem.value === sortValue && !selectedSortItem.ascending && (
+            <FaLongArrowAltDown></FaLongArrowAltDown>
+          )}
+        </div>
+      ),
+    };
+  });
+
+  return (
+    <div className={styles.Tasklist}>
+      <div className={selectedUserTaskID ? `${styles.list} selected` : styles.list}>
+        <div className={styles.actionWrapper}>
+          {selectedUserTaskID && !breakpoint.xl ? (
+            <Button
+              className={styles.backButton}
+              icon={<IoArrowBack></IoArrowBack>}
+              onClick={() => {
+                setSelectedUserTaskID(null);
+              }}
+            ></Button>
+          ) : (
+            <Space.Compact className={styles.dropdownWrapper}>
+              <Dropdown
+                open={filterDropdownOpen}
+                onOpenChange={(nextOpen: boolean, info: { source: string }) => {
+                  if (info.source === 'trigger' || nextOpen) {
```

Comment:
I implemented a custom opening mechanism for this dropdown because I didn't want it to close if an user clicks inside of the menu (e.g. for filtering tasks). This is explained [here](https://ant.design/components/dropdown#components-dropdown-demo-overlay-open)